File: sdks/flutterflow/test/emdash_store_parity_test.dart
import 'dart:convert';
import 'dart:io';

import 'package:flutter_test/flutter_test.dart';
import 'package:superboard_flutterflow/superboard_flutterflow.dart';

void main() {
  test('FlutterFlow round-trips the shared fixture with the real SDK model', () {
    final fixture =
        jsonDecode(
              File(
                '../../packages/contracts/fixtures/emdash-store-parity/v1.json',
              ).readAsStringSync(),
            )
            as Map<String, dynamic>;
    final event = SuperBoardCustomerEvent.fromJson(
      (fixture['customer_event'] as Map).cast<String, dynamic>(),
    );
    expect(event.toJson(), fixture['customer_event']);
    expect(
      (fixture['aliases'] as Map<String, dynamic>)['projectId'],
      fixture['instance_id'],
    );
  });
}

File: tests/checks/sdks/javascript/emdash-store-parity.test.js
import assert from "node:assert/strict";
import { readFileSync } from "node:fs";
import { test } from "node:test";

import SuperBoardAPIService from "../../../../sdks/javascript/src/superboard_api_service.js";
import SuperBoardContext from "../../../../sdks/javascript/src/superboard_context.js";

const fixture = JSON.parse(
	readFileSync(
		new URL("../../../fixtures/contracts/emdash-store-parity/v1.json", import.meta.url),
		"utf8",
	),
);

test("JavaScript sends the shared fixture through the real SDK encoder", () => {
	const calls = [];
	const service = new SuperBoardAPIService();
	service.apiService = { POST: (path, body) => calls.push({ path, body }) };
	SuperBoardContext.USER_IDENTIFIER = fixture.application_user.identifier;
	SuperBoardContext.USER_ATTRIBUTES = fixture.application_user.attributes;
	service.setUserAttributes(
		() => undefined,
		() => undefined,
	);
	assert.deepEqual(calls, [
		{
			path: SuperBoardAPIService.ENDPOINTS.USER_ATTRIBUTES,
			body: { sdk_identifier: "user-1", sdk_attributes: { active: true } },
		},
	]);
	assert.equal(fixture.aliases.projectId, fixture.instance_id);
	assert.equal(fixture.aliases.pid, fixture.instance_id);
});
